pkg/procfs/procfs.go

```diff
@@ -51,13 +51,8 @@ func FindPidByPodContainer(podUID, containerID string) (string, error) {
 				// See https://github.com/kubernetes/kubernetes/blob/2f3a4ec9cb96e8e2414834991d63c59988c3c866/pkg/kubelet/cm/cgroup_manager_linux.go#L81-L85
 				// Note that these identifiers are currently specific to systemd, however, this mounting approach is what allows us to find the containerized
 				// process.
-				//
-				// EG: /kubepods/burstable/pod31dd0274-bb43-4975-bdbc-7e10047a23f8/851c75dad6ad8ce6a5d9b9129a4eb1645f7c6e5ba8406b12d50377b665737072
-				//     /kubepods/burstable/pod{POD_ID}/{CONTAINER_ID}
-				//
-				// This "needle" that we look for in the mountinfo haystack should match one and only one container.
-				needle := path.Join(podUID, containerID)
-				if strings.Contains(root, needle) {
+				// EG: /kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod40934a46_d409_4f4a_bb8b_1ec0b0436320.slice/cri-containerd-afc114c69e71f18166abd63715398d7daa763c69a6454f71157272ab0bdca783.scope
+				if strings.Contains(root, podUID) && strings.Contains(root, containerID) {
```

Comment:
Which setup have you tested this with?
I also found that I needed to fix this when working on my PR: https://github.com/iovisor/kubectl-trace/pull/205/files#diff-3fc753f2ae105df1b499d656901818c3aae006a40a0d807d4871520a19d612a5 

I found `containerID` contained `-`s in place of `_`s in the cgroup name.
i.e. in your example, I'd expect `containerID` to be `"40934a46-d409-4f4a-bb8b-1ec0b0436320"`.
It looks like that was also mentioned in the description of #174 . (I was using kind v0.19.x on an ubuntu 22.04 VM, _not sure the host matters though_)

You could also feel inspired to add a unit test [like I did](https://github.com/iovisor/kubectl-trace/pull/205/files#diff-ef3fedfda983b569a7bc67946f6d0018c1e6da2c5c14650029fd534eabf0e193)..., but I'm not from the iovisor group, just a fellow attempting contributor.